File: code/online_challenges/src/project_euler/problem_012/problem_012.cpp
/*

// online challenges | project euler | problem 012 | problem 012 | C++
// Part of Cosmos by OpenGenus Foundation

*/


#include <iostream>
#include <cmath>

int main()
{
    int divisorCount = 0;
    int triangleNumberIndex = 0;
    int triangleNumber = 0;

    while (divisorCount < 500)
    {
        divisorCount = 0;
        ++triangleNumberIndex;
        triangleNumber += triangleNumberIndex;
        for (int i = 1; i < std::sqrt(triangleNumber) + 1; ++i)
            if (triangleNumber % i == 0)
                divisorCount += (i * i == triangleNumber) ? 1 : 2;
    }

    std::cout << triangleNumber << "\n";
    return 0;
}

File: code/online_challenges/src/project_euler/problem_012/problem_012.py
import math


def main():
    i = 1
    triangle_number = 0
    divisor_count = 0

    while divisor_count < 500:
        triangle_number += i
        i += 1
        divisor_count = 0

        for z in range(1, int(math.sqrt(triangle_number))):
            if triangle_number % z == 0:
                divisor_count += 2

    print(triangle_number)


if __name__ == "__main__":
    main()
